### PR TITLE
Additional python tools

### PR DESCRIPTION
**Pull request by abaillod**
Hi!

This branch started as a fix to an issue in the python tools (how Nvol was managed in spec_namelist), and then I started adding some useful routines to SPECout. 

I also made a small fix in SPEC final diagnostics. I thought it could be useful to have the Btemn, Btomn, ... harmonics on the computational boundary, so I changed the logic in line 879 or src/xspech.f90.

I think we can safely merge this, I did not touch the core of SPEC.

### CODE REVIEW

**Comment by abaillod** [comment]
Hi,

Could we merge these small changes? These are mainly new routines for the python tools.

The automatic tests seem to be failing because the docker container did not install scipy. I don't know how to fix this...

**Comment by zhucaoxiang** [comment]
I think I fixed the scipy error. But now we got CI errors like
```
ERROR: Btemn , element average difference =  9.07119E-02
ERROR: Btomn , element average difference =  2.14677E-02
ERROR: Bzemn , element average difference =  1.10917E+01
ERROR: Bzomn , element average difference =  1.06942E+00
```

**Comment by abaillod** [comment]
Thank you!

I will look into the Btemn error.

**Comment by abaillod** [comment]
I solved the `Btemn` bug; I think it is ready to be merged.

File: src/bfield.f90 [hunk 1]
```diff
@@ -166,7 +166,7 @@ subroutine bfield( zeta, st, Bst )
    write(ounit,'("bfield : ",f10.2," : lvol=",i3," ; zeta="es23.15" ; (s,t)=("es23.15" ,"es23.15" ) ; B^z="es23.15" ;")') &
                              cput-cpus, lvol,        zeta,             st(1:2),                       dBu(3)
 
-   FATAL( bfield, abs(dBu(3)).lt.vsmall, field is not toroidal )
+   !FATAL( bfield, abs(dBu(3)).lt.vsmall, field is not toroidal )
```
Comment (zhucaoxiang) [inline src/bfield.f90:169]: @abaillod Any particular reason to comment this line? Though I do agree it is annoying sometimes.

Comment (abaillod) [reply src/bfield.f90:169]: Yes - it is indeed annoying, as it makes a SPEC calculation crash. I don't think there is any use to this anyway... Maybe a warning would be enough?

Comment (zhucaoxiang) [reply src/bfield.f90:169]: Yes, let's make it a warning.

**Comment by zhisong** [comment]
This is OK but your RK4 will fail. The right-hand side will be e.g. Bt/Bz -> infinity. Maybe there is a way to quit the integration but not throw an error. Anyway, unless you are dealing with RFP, if the problem is triggered it means the calculation is really fishy.

**Comment by zhisong** [comment]
You should also change the version of py_spec or it will not get updated on PYPI. You should find the version number in `__init__.py`.

**Comment by abaillod** [comment]
> This is OK but your RK4 will fail. The right-hand side will be e.g. Bt/Bz -> infinity. Maybe there is a way to quit the integration but not throw an error. Anyway, unless you are dealing with RFP, if the problem is triggered it means the calculation is really fishy.

I agree. Maybe the best is to keep the `FATAL`statement, and change in a later commit the behavior of the integrator when this test is triggered. I will revert it.

**Comment by abaillod** [comment]
If there are no more comments, I will merge this branch by the end of the day. Thank you for your reviews!

**Pull request merged**